internal/controller/warehouses/git.go

```diff
@@ -94,13 +94,20 @@ func (r *reconciler) selectCommitMeta(
 	if sub.CommitSelectionStrategy == "" {
 		sub.CommitSelectionStrategy = kargoapi.CommitSelectionStrategyNewestFromBranch
 	}
+	// for cases when ScanPaths and/or IgnorePaths filters are used we want to have 2
+	// commits cloned to be able to get actual DiffPaths of the last commit to compare
+	// against the filters
+	var cloneDepth uint8 = 1
+	if sub.ScanPaths != nil || sub.IgnorePaths != nil {
```

Comment:
```suggestion
	if len(sub.ScanPaths) != 0 || len(sub.IgnorePaths) != 0 {
```

This will avoid unnecessarily increasing the clone depth if one of these is non-nil, but still empty.